### PR TITLE
feat: adds support for `x-www-form-urlencoded`

### PR DESCRIPTION
**Pull request by kevinperaza**
## Description

- feat: adds support for `x-www-form-urlencoded`
- adds example for creating a card in stripe

## Testing required outside of automated testing?

- [ ] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: IntegrationTester/IntegrationTester/HttpClientViewController.swift [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        print("expirationDateTextField.text: \(expirationDateTextField.text)")
+        print("cvcTextField.text: \(cvcTextField.text)")
+    }
+    
+    @IBAction func createPaymentMethod(_ sender: Any) {
+        let body: [String: Any] = [
+            "type": "card",
+            "billing_details": [
+                "name": "Peter Panda"
+            ],
```
Comment (brigonzalez) [inline IntegrationTester/IntegrationTester/HttpClientViewController.swift:40]: I think this gets stripped off?

File: BasisTheoryElements/Sources/BasisTheoryElements/HttpClientHelpers.swift [hunk 2]
```diff
@@ -34,7 +91,10 @@ struct HttpClientHelpers {
                 return
             }
             
-            let httpBody = try! JSONSerialization.data(withJSONObject: mutablePayload, options: [])
+            let contentType = config?.headers["Content-Type"] ?? "application/json"
```
Comment (brigonzalez) [inline BasisTheoryElements/Sources/BasisTheoryElements/HttpClientHelpers.swift:94]: you're looking at the headers here for how to map the object to the body. but we should probably also set the headers here to "application/json" here if it's not set as well

Comment (kevinperaza) [reply BasisTheoryElements/Sources/BasisTheoryElements/HttpClientHelpers.swift:94]: Updated

File: IntegrationTester/UnitTests/HttpClientTests.swift [hunk 3]
```diff
@@ -52,6 +52,45 @@ final class HttpClientTests: XCTestCase {
...
+            ],
+            "array": [1, 2, 3]
+        ] as [String : Any]
+        
+        let config = Config(headers: [
+            "Content-Type": "application/x-www-form-urlencoded",
+        ])
+        
+        let postHttpClientExpectation = self.expectation(description: "POST HTTP client")
+        BasisTheoryElements.post(url: "https://echo.basistheory.com/anything", payload: payload, config: config) { request, data, error in
```
[the remark this answers is not in the document]
Comment (kevinperaza) [reply IntegrationTester/UnitTests/HttpClientTests.swift:72]: Is there a way to access the config once it's passed to `BasisTheoryElements.post`? For Android, we used `MockWebServer` to intercept requests and assert against them but I didn't find anything worth using for iOS and surfacing some of these internals for just testing feels weird

File: BasisTheoryElements/Sources/BasisTheoryElements/HttpClientHelpers.swift [hunk 4]
```diff
@@ -34,11 +91,18 @@ struct HttpClientHelpers {
...
+            
+            let httpBody = mapObjToRequestBody(contentType: contentType, obj: mutablePayload)
+            
             request.httpBody = httpBody
         }
         
         if let config = config {
+            if(config.headers["Content-Type"] == nil) {
+                request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+            }
```
Comment (kevinperaza) [inline BasisTheoryElements/Sources/BasisTheoryElements/HttpClientHelpers.swift:104]: Default content-type header to application/json if it's not present in the config

**Review by lcschy — APPROVED** [review]

**Pull request merged**